allow Vue nodes to be resized from all 4 corners

christian-byrne opened the pull request:
## Summary

Enables Vue nodes to resize from all four corners and consolidated the interaction pipeline.

## Changes

- **What**: Added four-corner handles to `LGraphNode`, wired them through the refactored `useNodeResize` composable, and centralized the math/preset helpers under `interactions/resize/` with cleaner pure functions and lint-compliant markup.

## Review Focus

Corner-to-corner resizing accuracy (position + size), pinned-node guard preventing resize start, and snap-to-grid behavior at varied zoom levels.

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-6187-allow-Vue-nodes-to-be-resized-from-all-4-corners-2936d73d365081c8bf14e944ab24c27f) by [Unito](https://www.unito.io)

christian-byrne committed:
resize from all 4 corners

christian-byrne committed:
simplify tests

DrJKL commented on src/renderer/extensions/vueNodes/components/LGraphNode.vue:
```diff
@@ -243,8 +247,12 @@ onErrorCaptured((error) => {
+  position,
+  size,
+  zIndex,
+  moveTo: moveNodeTo
```
Should the name be changed inside `useNodeLayout`?

DrJKL commented on src/renderer/extensions/vueNodes/interactions/resize/resizeMath.ts:
```diff
@@ -0,0 +1,132 @@
+function clampToMinSize(size: Size, minSize: Size): Size {
+  return {
+    width: clamp(size.width, minSize.width, Number.POSITIVE_INFINITY),
+    height: clamp(size.height, minSize.height, Number.POSITIVE_INFINITY)
```
I'd just use `Math.max` here.

christian-byrne committed:
[refactor] use Math.max instead of clamp for min size constraints - addresses review feedback

Co-authored-by: DrJKL <DrJKL@users.noreply.github.com>

christian-byrne committed:
[refactor] extract validated resize state to local variables - addresses review feedback

Co-authored-by: DrJKL <DrJKL@users.noreply.github.com>

christian-byrne committed:
[refactor] extract CornerResizeHandle type and annotate array - addresses review feedback

Co-authored-by: DrJKL <DrJKL@users.noreply.github.com>

christian-byrne committed:
[refactor] rename moveTo to moveNodeTo in useNodeLayout - addresses review feedback

Co-authored-by: DrJKL <DrJKL@users.noreply.github.com>

christian-byrne committed:
[style] use Tailwind group pattern for resize handle visibility - addresses review feedback

Co-authored-by: DrJKL <DrJKL@users.noreply.github.com>

christian-byrne committed:
[refactor] rename factor variables to multiplier for clarity - addresses review feedback

christian-byrne committed:
[a11y] add aria-label and role to resize handles - addresses review feedback

christian-byrne committed:
[test] add edge case tests for resize math - addresses review feedback

christian-byrne committed:
[fix] keep resize handles invisible - only show cursor change

Handles should remain opacity-0 with only the cursor indicating
resize capability. Removed group-hover visibility classes.

actions-user committed:
[automated] Apply ESLint and Prettier fixes

christian-byrne commented on src/renderer/extensions/vueNodes/interactions/resize/useNodeResize.ts:
```diff
@@ -2,127 +2,127 @@ import { useEventListener } from '@vueuse/core'
+        size: Size
+        position: Point
+      })
+    | null
```
Extracted to local variables in 4e1168f

christian-byrne commented on src/renderer/extensions/vueNodes/components/LGraphNode.vue:
```diff
@@ -282,19 +290,66 @@ onMounted(() => {
 })
 
+const baseResizeHandleClasses =
+  'absolute h-3 w-3 opacity-0 pointer-events-auto focus-visible:outline focus-visible:outline-2 focus-visible:outline-white/40'
```
Implemented in 68255c9 (removed in feda4d9 to keep handles invisible)

christian-byrne commented on src/renderer/extensions/vueNodes/interactions/resize/resizeMath.ts:
```diff
@@ -0,0 +1,132 @@
+  currentPointer: Point,
+  scale: number
+): Point {
+  const safeScale = scale === 0 ? 1 : scale
```
Skipping - only used once, extraction wouldn't add value

christian-byrne commented on tests-ui/tests/renderer/extensions/vueNodes/interactions/resize/resizeMath.test.ts:
```diff
@@ -0,0 +1,81 @@
+
+    expect(delta).toEqual({ x: 50, y: 30 })
+  })
+})
```
Added 6 edge case tests covering zero scale, negative/extreme deltas, and boundary conditions in 27222a0

christian-byrne commented on src/renderer/extensions/vueNodes/components/LGraphNode.vue:
```diff
@@ -282,19 +290,66 @@ onMounted(() => {
 
+const baseResizeHandleClasses =
+  'absolute h-3 w-3 opacity-0 pointer-events-auto focus-visible:outline focus-visible:outline-2 focus-visible:outline-white/40'
+const POSITION_EPSILON = 0.01
```
Skipping - unclear rationale for the threshold
